rs/boundary_node/rate_limits/canister/canister.rs

```diff
@@ -100,9 +105,9 @@ fn post_upgrade(init_arg: InitArg) {
 fn get_config(version: Option<Version>) -> GetConfigResponse {
     let caller_id = ic_cdk::api::caller();
     let response = with_canister_state(|state| {
-        let access_resolver = AccessLevelResolver::new(caller_id, state.clone());
-        let formatter = ConfigConfidentialityFormatter;
-        let getter = ConfigGetter::new(state, formatter, access_resolver);
+        let getter = ConfigGetter::new(state.clone());
+        let access_resolver = AccessLevelResolver::new(caller_id, state);
```

Comment:
This introduces another "layer" but doesn't change it. I understood their ask more like:
```
let access_resolver = AccessLevelResolver::new(caller_id, state.clone());
let getter = ConfigGetter::new(state.clone());
let getter = WithAuthorization::new(adder, access_resolver);
getter.get(&version)
```
And the authorization layer has something like:
```
let access_level = self.access_resolver.get_access_level();
let redacted = !(access_level == AccessLevel::FullAccess || access_level == AccessLevel::FullRead);
getter.get(&version, redacted)
```
redacted should be optional and getter.get should always default to false if nothing is specified. At the moment, I feel it is even less secure as now the getter.get by default exposes all the information and if someone removes the WithConfigConfidentialityFormatting, everything goes out.

Reply:
I think You understood the comment correctly. But IMO, this is not how it should be done. 

As was also correctly noted, from the responsibility perspective `ConfigGetter` should just return the config. Redacting should be done by another layer (wrapper). Hence, `ConfigGetter`'s `get` method shouldn't accept anything but version. Now `WithConfigConfidentialityFormatting` is architecturally aligned with `WithAuthorization`. That's what IMO was important to achieve.

> At the moment, I feel it is even less secure as now the getter.get by default exposes all the information and if someone removes the WithConfigConfidentialityFormatting, everything goes out.

Well the same critics can be applied to `WithAuthorization`, of course one shouldn't remove these wrappers.
This is why i have an important [test](https://github.com/dfinity/ic/pull/3222/files#diff-ce2f2f97272ed579da4f5db0538dbb86b72e0e01523a2601d86ee82f39736ec6R204) and [assertions](https://github.com/dfinity/ic/pull/3222/files#diff-ce2f2f97272ed579da4f5db0538dbb86b72e0e01523a2601d86ee82f39736ec6R290) to protect against accidental changes. Let me know if my explanation sounds reasonable.